### PR TITLE
Add `batch` utility

### PR DESCRIPTION
**Pull request by RobinMalfait**

### CODE REVIEW

**Comment by bjuppa** [comment]
I like this direction @RobinMalfait ! The `wait` function is a great way to make intention clear in a program.

As described in the readme update, this `eager()` function processes batches of a fixed size, so I suggest actually calling it `batch()`, to clarify the grouped-processing behaviour. Does that make sense? Or would the `batch` name intuitively indicate some other collection-operation?

Then if we proceed to add more functions in this style, what I called `eager()` (in the draft in #4) could perhaps be named `parallel()` instead. That one would start processing a new `next()` as soon as any slot is free in the "buffer", not waiting for the full batch to complete.
And then add `race()` too, if we want a variant that mess with collection order (that will be a challenge to write some solid tests for 😄).

File: src/eager.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+        while (state.more) {
+          for (let i = 0; i < size; i++) {
+            const { value, done } = iterator.next()
+            if (done) state.more = false
+            if (!done) buffer.push(value)
+          }
+
+          yield* buffer.splice(0)
+        }
```
Comment (bjuppa) [inline src/eager.ts]: Regarding async iterables here, I think there's a good case to let them through and process them in the same manner!

On line `19` (when pulling out `next()`), that would return a promise if `data` is asynciterable, all that's needed to do in that case is await it, before extracting `value` and `done`. Everything else could stay the same, I think... `await` on a non-promise would just resolve right away, am I right?

File: src/eager.test.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+it('should be possible to eager load an API call for example (from sync -> async)', async () => {
+  function fetch(path: string) {
+    return new Promise(resolve => setTimeout(resolve, 50, path))
+  }
+
+  const program = pipe(
+    map(id => `/user/${id}`),
+    map<string, unknown>(path => fetch(path)),
+    eager(10),
```
Comment (bjuppa) [inline src/eager.test.ts]: If this test is processing 10 items, what about testing in batches of `5` or `4`, just to spice it up a bit? 😄

Comment (RobinMalfait) [reply src/eager.test.ts]: Oh yes I agree. This allows me to implicitly test that it is doing the batching correctly!

**Comment by RobinMalfait** [comment]
*knock knock* is this thing still on?

Welp, it took a while, but let's get this thing merged!

**Pull request merged**